### PR TITLE
:bug: Remove Version from references when comparing MachineTemplateSpec

### LINKED ISSUE #2183: Change MachineDeployment logic that determines if a new MachineSet is needed to exclude the version portion of ObjectReferences
**User Story**

As a user I would like to be able to change the API version of the bootstrap and infrastructure references in my MachineDeployment's template without triggering a rolling update, so I can update to newer API versions instead of staying on older, soon-to-be out of date ones.

**Detailed Description**

The following code is used to determine if the machine template in the MachineDeployment matches the template in the MachineSets:

https://github.com/kubernetes-sigs/cluster-api/blob/403fbd8e8ff85e3c9001e59ccc79cc6b095e521c/controllers/mdutil/util.go#L394-L422

Now that we've moving forward with v1alpha3, and we're supporting webhook conversions to/from v1alpha2, we may encounter existing MachineDeployments and MachineSets that have bootstrap and/or infrastructure references to an older API version (e.g. v1alpha2). When the management cluster is upgraded to v1alpha3, the user may decide they want to update their bootstrap and infrastructure references to use v1alpha3 (or it may be done automatically - see #1938).

If the APIVersion of a bootstrap or infrastructure reference changes, this should **not** trigger a rolling update of the MachineDeployment. That would cause machines to be deleted and replaced, when it's extremely unlikely the user has any need or desire for this churn.

Because the EqualIgnoreHash function is working on deep copies of the templates, we can modify it to adjust the APIVersion field of the bootstrap and infrastructure references to be just the API group, and exclude the version:

```go
// if config ref != nil, etc...
bootstrapGV1, _ := schema.ParseGroupVersion(t1Copy.Spec.Bootstrap.ConfigRef.APIVersion)
bootstrapGroup1 := bootstrapGV1.Group
t1Copy.Spec.Bootstrap.ConfigRef.APIVersion = bootstrapGroup1
// do the same for bootstrapGV2, and also for the 2 infra refs
```

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

/kind feature
/priority important-soon
/milestone v0.3.0

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2183

/assign @ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
Can you add a test that has templates that are identical except for bootstrap and infra GVs?

**Comment by vincepri** [comment]
yeah sec

File: controllers/mdutil/util_test.go [hunk 1]
```diff
@@ -122,6 +122,12 @@ func TestEqualIgnoreHash(t *testing.T) {
 			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{DefaultMachineDeploymentUniqueLabelKey: "value-2", "something": "else"}),
 			true,
 		},
+		{
+			"Same spec, the former doesn't have machine-template-hash label",
+			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{"something": "else"}),
+			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{DefaultMachineDeploymentUniqueLabelKey: "value-2", "something": "else"}),
+			true,
+		},
```
Comment (detiber) [inline controllers/mdutil/util_test.go]: Should there also be a test case for when the api versions of objectReferences differ?

Comment (vincepri) [reply controllers/mdutil/util_test.go]: Yep added both

**Pull request merged**